cvmfs/clientctx.h

```diff
@@ -56,9 +56,13 @@ class ClientCtx {
  */
 class ClientCtxGuard {
  public:
-  ClientCtxGuard(uid_t uid, gid_t gid, pid_t pid) :
-      set_on_construction_(false) {
-      // Implementation guarantees old_ctx is not null.
+  ClientCtxGuard(uid_t uid, gid_t gid, pid_t pid)
+    : set_on_construction_(false)
+    , old_uid_(0)
```

Comment:
uid, gid, and pid should be initialized to -1 to indicate that they are invalid.